### PR TITLE
Tests for FileWatcher

### PR DESCRIPTION
**Pull request by krnowak**
This is based on commits from PR #2855, so this PR is blocked until the prerequisite is merged.

This PR basically adds more thorough tests for the FileWatcher utility.

I'm not sure if you will want to take the last commit (about debugging), because it adds a lot of code just for the nicer debug output. It helped me a lot though when I was figuring out bugs in FileWatcher.

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/sup/src/manager/mod.rs [hunk 1]
```diff
@@ -882,6 +896,22 @@ impl Manager {
...
+    fn update_peers_from_watch_file(&mut self) -> Result<()> {
+        if !self.butterfly.need_peer_seeding() {
+            return Ok(());
+        }
+        match self.peer_watcher {
+            None => Ok(()),
+            Some(ref watcher) => {
+                if watcher.has_fs_events() {
+                    let members = watcher.get_members()?;
+                    self.butterfly.member_list.set_initial_members(members);
```
Comment (christophermaier) [inline components/sup/src/manager/mod.rs]: When the peer file changes, how do the new peers get added into the list of "active" members? I don't see where the peer list ever gets read again after the initial startup.

**Comment by christophermaier** [comment]
We're going to go ahead and merge this.

@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/285314385).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/285314385).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

File: components/sup/src/manager/mod.rs (hunk 1 above)
Comment (krnowak) [reply components/sup/src/manager/mod.rs]: The `--peer-watch-file` flag works a bit like the `--peer` flag - it sets the initial members list and then the butterfly outbound thread will ping each member from that list. I guess that the inbound thread will handle the pong reply and initiate the move of the member from the "initial" list to the "active" list.

Also, like in the `--peer` flag, once we are a part of the ring, we don't really care about this anymore.

**Pull request merged**